### ISSUE TITLE
Ability to see all running jobs for similar method  

### ISSUE THREAD

**Issue by ibrahim12** [rq/rq#1117]
Currently there is no way to search or query all jobs for a similar method other then fetch all jobs and map in python. 

What about having that functionality ? May be storing job ids by method.

Code example
```
def add(a, b):
   return a+ b

def minus(a, b):
  return a + b

Jobs for add

add(1, 2)
add(2, 3)
add(4, 5)

Jobs for minus

minus(2, 3)
minus(4, 5)
minus(6, 7)
```

**Comment by selwin** [comment]
No, at the moment we don't keep track of job types. This would be something that I'm looking to add in the future.

**Comment by ibrahim12** [comment]
I would be happy to discuss about the changes required and work on it if you are interested

**Comment by selwin** [comment]
My initial thinking is that we can have something like a `JobTypeRegistry` which contains job IDs and their expiration time.

Similar to `StartedJobRegistry` and other registry types, we'll have one Registry for each type of job and queue combination. So if you have jobs `foo` and `bar`, there will be two registries, each containing job IDs and their expiration times (if any).

**Comment by ibrahim12** [comment]
So if we have n job types, the we would require n (registry + queue combination). 

Do you think this will scale with larger job types ?
